services/ros/ros-version.service.js

```diff
@@ -0,0 +1,111 @@
+import gql from 'graphql-tag'
+import Joi from 'joi'
+import { renderVersionBadge } from '../version.js'
+import { GithubAuthV4Service } from '../github/github-auth-service.js'
+import { NotFound } from '../index.js'
+import { parseReleaseVersionFromDistro } from './ros-version-helpers.js'
+
+const tagsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      refs: Joi.object({
+        edges: Joi.array()
+          .items({
+            node: Joi.object({
+              name: Joi.string().required(),
+            }).required(),
+          })
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const contentSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      object: Joi.object({
+        text: Joi.string().required(),
+      }).allow(null),
+    }).required(),
+  }).required(),
+}).required()
+
+export default class RosVersion extends GithubAuthV4Service {
+  static category = 'version'
+
+  static route = { base: 'ros/v', pattern: ':distro/:packageName' }
+
+  static examples = [
+    {
+      title: 'ROS Package Index',
+      namedParams: { distro: 'humble', packageName: 'vision_msgs' },
+      staticPreview: {
+        ...renderVersionBadge({ version: '4.0.0' }),
+        label: 'humble',
+      },
+    },
+  ]
+
+  async handle({ distro, packageName }) {
+    const tagsJson = await this._requestGraphql({
+      query: gql`
+        query ($refPrefix: String!) {
+          repository(owner: "ros", name: "rosdistro") {
+            refs(
+              refPrefix: $refPrefix
+              first: 30
+              orderBy: { field: TAG_COMMIT_DATE, direction: DESC }
+            ) {
+              edges {
+                node {
+                  name
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: { refPrefix: `refs/tags/${distro}/` },
+      schema: tagsSchema,
+    })
+
+    // Filter for tags that look like dates: humble/2022-06-10
+    const tags = tagsJson.data.repository.refs.edges
+      .map(edge => edge.node.name)
+      .filter(tag => /^\d+-\d+-\d+$/.test(tag))
```

Comment:
Is what we actually ideally want here

- the tag corresponding to the most recent commit date (where the tag name looks like a date), or
- the tag with the most recent date in the tag name.

In general, I'd hope those will be the same, but making the assumption we want to code defensively here.